org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/instance/InstanceValidator.java

```diff
@@ -3472,6 +3480,101 @@ private void buildIdentifierExpression(ElementDefinition ed, StringBuilder expre
+      throw new DefinitionException(context.formatMessage(I18nConstants.UNSUPPORTED_IDENTIFIER_PATTERN_PROPERTY_NOT_SUPPORTED_FOR_DISCRIMINATOR_FOR_SLICE, discriminator, ed.getId(), name.fhirType(), "given"));
+    }
+    if (name.hasPrefix()) {
+      throw new DefinitionException(context.formatMessage(I18nConstants.UNSUPPORTED_IDENTIFIER_PATTERN_PROPERTY_NOT_SUPPORTED_FOR_DISCRIMINATOR_FOR_SLICE, discriminator, ed.getId(), name.fhirType(), "prefix"));
+    }
+    if (name.hasSuffix()) {
+      throw new DefinitionException(context.formatMessage(I18nConstants.UNSUPPORTED_IDENTIFIER_PATTERN_PROPERTY_NOT_SUPPORTED_FOR_DISCRIMINATOR_FOR_SLICE, discriminator, ed.getId(), name.fhirType(), "suffix"));
+    }
+    if (name.hasPeriod()) {
+      throw new DefinitionException(context.formatMessage(I18nConstants.UNSUPPORTED_IDENTIFIER_PATTERN_PROPERTY_NOT_SUPPORTED_FOR_DISCRIMINATOR_FOR_SLICE, discriminator, ed.getId(), name.fhirType(), "period"));
+    }
+    if (first) {
+      throw new DefinitionException(context.formatMessage(I18nConstants.UNSUPPORTED_IDENTIFIER_PATTERN_NO_PROPERTY_NOT_SUPPORTED_FOR_DISCRIMINATOR_FOR_SLICE, discriminator, ed.getId(), name.fhirType()));
+    }
+
+    expression.append(").exists()");
+  }
+
+  private void buildAddressExpression(ElementDefinition ed, StringBuilder expression, String discriminator, Address add) throws DefinitionException {
+    if (add.hasExtension()) {
+      throw new DefinitionException(context.formatMessage(I18nConstants.UNSUPPORTED_IDENTIFIER_PATTERN__EXTENSIONS_ARE_NOT_ALLOWED__FOR_DISCRIMINATOR_FOR_SLICE_, discriminator, ed.getId()));
+    }
+    boolean first = true;
+    expression.append(discriminator + ".where(");
+    if (add.hasUse()) {
+      first = false;
+      expression.append("use = '" + add.getUse().toCode() + "'");
+    }
+    if (add.hasType()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("type = '" + add.getType().toCode() + "'");
+    }
+    if (add.hasText()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("text = '" + add.getText() + "'");
+    }
+    if (add.hasCity()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("city = '" + add.getCity() + "'");
+    }
+    if (add.hasDistrict()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("district = '" + add.getDistrict() + "'");
+    }
+    if (add.hasState()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("state = '" + add.getState() + "'");
+    }
+    if (add.hasPostalCode()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("postalCode = '" + add.getPostalCode() + "'");
+    }
+    if (add.hasCountry()) {
+      if (first) first = false; else expression.append(" and ");
+      expression.append("country = '" + add.getCountry() + "'");
+    }       
```

Comment:
Why the strange if/else formatting in this file? would be easier to use the ternary operator